First level upgrades

MartinPerez opened the pull request:
This PR handles several light changes to first level models that were being mixed with the second level Pr and other small but useful modifications.

1) Progress messages have been refined and verbose documentation adjusted accordingly.
2) paradigms has been changed to events to comply with BIDS terminology
3) a TR warning is given in case the TR value is suspiciously high, likely given in ms instead of s.
4) Patsy text formulas added in contrasts
5) Memcache of contrasts dropped. It easily introduced bugs when repeated across different models and is unnecessary from the computational point of view.
6) Error check of column names of design matrix passed to design matrix creation.

tests updated accordingly. This PR requires merge of #88 to install new dependencies in travis. (Patsy).
Will rebase to master once #88 is merged.

bthirion commented on nistats/first_level_model.py:
```diff
@@ -246,10 +250,12 @@ def __init__(self, t_r=None, slice_time_ref=0., hrf_model='glover',
                  minimize_memory=True):
         # design matrix parameters
         self.t_r = t_r
+        if t_r > 10:
```
Should we take care of that ? In which case is such a check relevant ?
I'm afraid of adding such overhead in the codebase.

MartinPerez replied:
I thought it could be a very nice feature, its an inexpensive check for a very common mistake handling the time unit of t_r, moreover the impact of this type of mistake is very important.

bthirion replied:
yes, but following this logic we could propose this type if check for many possible combinations of arguments. I really don't want to go into that direction now, so if you don't remove, I'd rather not introduce this feature.

bthirion commented on nistats/first_level_model.py:
```diff
@@ -326,44 +335,54 @@ def fit(self, run_imgs, paradigms=None, confounds=None,
-        if paradigms is not None:
-            paradigms = _check_run_tables(run_imgs, paradigms, 'paradigms')
+        # check the number of events and confound files match number of runs
+        # Also check events and confound files can be loaded as DataFrame
```
check that...

bthirion commented on nistats/first_level_model.py:
```diff
@@ -326,44 +335,54 @@ def fit(self, run_imgs, paradigms=None, confounds=None,
                 standardize=self.standardize, mask_strategy='epi',
                 t_r=self.t_r, memory=self.memory,
-                verbose=max(0, self.verbose - 1),
+                verbose=max(0, self.verbose - 2),
```
One word of explanation on reducing verbosity by 2 ?

MartinPerez replied:
check the verbosity documentation. Everything is there.

MartinPerez replied:
I mean on FirstLevelModel.

bthirion replied:
OK.

bthirion commented on nistats/first_level_model.py:
```diff
@@ -464,23 +503,27 @@ def compute_contrast(self, contrast_def, contrast_name=None,
+        design_info = DesignInfo(self.design_matrices_[0].columns.tolist())
+        for cidx, con in enumerate(con_vals):
+            if not isinstance(con, np.ndarray):
+                con_vals[cidx] = design_info.linear_constraint(con).coefs
```
This feature is very important, it has to be illustrated in an example.

MartinPerez replied:
I will add this as an issue. I think to be tackled after webpage docs with sphinx and circle ci are implemented.

bthirion commented on nistats/first_level_model.py:
```diff
@@ -433,27 +469,30 @@ def fit(self, run_imgs, paradigms=None, confounds=None,
 
-    def compute_contrast(self, contrast_def, contrast_name=None,
-                         stat_type=None, output_type='z_score'):
+    def compute_contrast(self, contrast_def, stat_type=None,
```
the examples still use `contrast_name`
Please update them

bthirion replied:
This tells us that we should now really add circleCI !

MartinPerez replied:
agreed, I will move up in the roadmap implementing sphinx and circleCI.

bthirion commented:
I have some minor requests --mostly related to the examples-- but the PR is good overall.

MartinPerez commented:
Updated all examples and fixed comments. Opened an issue for the patsy contrasts example to be done after sphinx and circle ci implementation. Waiting for merge of dependencies_update to rebase this PR, otherwise tests will fail from lack of patsy installation.

bthirion commented on nistats/first_level_model.py:
```diff
@@ -412,17 +431,34 @@ def fit(self, run_imgs, paradigms=None, confounds=None,
             if self.memory is not None:
-                mem_glm = self.memory.cache(run_glm)
+                arg_ignore = ['n_jobs']
+                mem_glm = self.memory.cache(run_glm, ignore=arg_ignore)
```
Probably no need to introduce arg_ignore explicilty.

MartinPerez replied:
why not? its possible recomputation would take place if the number of jobs is changed, or I misunderstand the meaning of the ignore parameter.

MartinPerez replied:
ahh sorry, you mean to create the var arg_ignore. Ok.

bthirion commented:
All right, we're almost there.

MartinPerez commented:
@bthirion took care of new comments and rebased, tests passed. Nonetheless I noticed in a previous commit that assert_array_almost_equal to 5 decimals would fail randomly. It reminds me of the issue raised in #77. Anyway this intermitent problem was with the scaling function which is not touched by this PR.

bthirion commented:
LGTM, merging.
